File: stock-market-service/src/main/java/rs/edu/raf/model/order/Order.java
package rs.edu.raf.model.order;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Inheritance(strategy = InheritanceType.JOINED)
public abstract class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Enumerated(EnumType.STRING)
    private OrderAction orderAction;
    private String ticker;
    private int quantity;
    private Long orderCreator;
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/model/order/Order.kt
package org.example.stockmarketservicekotlin.model.order

import jakarta.persistence.*

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
abstract class Order(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,
    @Enumerated(EnumType.STRING)
    val orderAction: OrderAction,
    val ticker: String,
    var quantity: Int,
    val orderCreator: Long
) {
}